File: tiq-e/array/7/main.py
from time import perf_counter
from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        for i in range(len(digits) - 1, -1, -1):
            if digits[i] != 9:
                digits[i] += 1
                break
            else:
                digits[i] = 0
                if i == 0:
                    digits.insert(0, 1)
        return digits


arr = [4, 1, 2, 1, 2]

start = perf_counter()
result = Solution().plusOne(arr)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/array/7/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut digits = digits;

        for i in (0..digits.len()).rev() {
            if digits[i] == 9 {
                digits[i] = 0;
            } else {
                digits[i] += 1;
                return digits;
            }
        }
        digits.insert(0, 1);

        digits
    }
}

fn main() {
    let arr = Vec::from([2, 3, 3, 2, 9]);

    let start = Instant::now();
    let result = Solution::plus_one(arr);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}